File: unit-2/Sprint-3/Day-1/Practice/Project 1/app/src/main/java/com/example/fragment1/MainActivity.java
package com.example.fragment1;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

public class MainActivity extends AppCompatActivity {

    private Button mBtnOne, mBtnTwo;
    private FragmentManager fragmentManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mBtnOne = findViewById(R.id.btnOne);
        mBtnTwo = findViewById(R.id.btnTwo);
        fragmentManager = getSupportFragmentManager();
        mBtnOne.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                FragmentA fragmentA = new FragmentA();
                fragmentTransaction.addToBackStack(null);
                fragmentTransaction.add(R.id.frame, fragmentA, "fragment A").commit();
            }
        });

        mBtnTwo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                FragmentB fragmentB = new FragmentB();
                fragmentTransaction.addToBackStack(null);
                fragmentTransaction.add(R.id.frame, fragmentB, "fragment B").commit();
            }
        });
    }
}

File: Callling_API/app/src/main/java/com/example/callling_api/MainActivity.kt
package com.example.callling_api

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), Interface {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val presenter = Presenter(getModel(), this)
        btnDownload.setOnClickListener {

            presenter.downloading()

        }


    }


    private fun getModel(): Model {
        return Model()
    }

    override fun download(download: String) {

        Glide.with(ivPic).load(download).into(ivPic)
    }
}